index.html

```diff
@@ -2694,12 +2711,196 @@ <h1>
         <h2>
 DID Resolution
         </h2>
+
+        <p>
+The <a>DID resolution</a> process takes a <a>DID</a> and resolution options as
+input, and produces a <a>DID document</a> and resolution metadata as output. The
+process utilizes the "Read" operation of the applicable <a>DID method</a>, as
+described in <a href="#read-verify"></a>. The <a>DID method</a>-specific details
+of how this process is accomplished is outside the scope of this specification.
+        </p>
+
+        <section>
+          <h3>
+Generic DID Resolution Algorithm
+          </h3>
+
+          <p>
+The generic DID resolution algorithm takes a <a>conforming DID</a>
+(<em>did</em>) and resolution options (<em>resolutionOptions</em>) as input and
+produces a <a>conforming DID Document</a> (<em>didDocument</em>), DID Document
+Metadata (<em>didDocumentMetadata</em>), and resolution metadata
+(<em>resolutionMetadata</em>) as output.
+          <p>
+
+          <p>
+Any errors encountered during the execution of this algorithm MUST be expressed
+using the <strong>error</strong> property in <em>resolutionMetadata</em>.
+          </p>
+
+          <p>
+When resolving a <a>DID</a> to a <a>DID document</a>, the following algorithm,
+or one producing an equivalent result, MUST be used:
+          </p>
+
+          <ol>
+            <li>
+Ensure that the <em>did</em> conforms to <a href="#generic-did-syntax"></a>.
+If the <em>did</em> does not conform, an <code>invalid did</code> error MUST be
+returned.
+            </li>
+            <li>
+Ensure that the each value in <em>resolutionOptions</em> conforms to  <a
+href="#resolution-options"></a>. If an option does not conform, an <code>invalid
+resolution option</code> error MUST be returned.
+            </li>
+            <li>
+Execute the <a>DID method</a>-specific "Read" operation, from the concrete
+<a>binding</a>, as described in <a href="#read-verify"></a> using the
+<em>did</em> and <em>resolutionOptions</em> as input. If the operation
+operation is unsuccessful, a <code>resolution failure</code> error MUST be
+returned.
+            </li>
+            <li>
+If successful, store the resulting <a>DID document</a>
+(<em>didDocument</em>), the DID Document Metadata
+(<em>didDocumentMetadata</em>), and the resolution metadata
+(<em>resolutionMetadata</em>).
+            </li>
+            <li>
+Ensure that <em>didDocument</em> is a <a>conforming DID document</a>. If the
+<em>didDocument</em> does not conform, an <code>invalid did document</code>
```

Comment:
Can we talk about what a "conforming" did document would be? For example, would `did:example:123 => { "@id":"did:example:456" }` be consider valid even though the returned did document is valid, but doesn't match the input identifier?

Reply:
> Can we talk about what a "conforming" did document would be or would that fall out of scope? 

The specification defines what a "conforming DID Document" is... so yes, it's in scope.

> For example, would did:example:123 => { "@id":"did:example:456" } be consider valid even though the returned did document is valid, but doesn't match the input identifier?

Per the current spec, it's invalid because of `@id` -- we don't allow that use because it goes against what the JSON-only folks want.

That the id doesn't map to the DID Document is valid as there is no normative requirement in the spec that states that the resulting DID Document MUST contain an `id` that matches the DID that was sent as the input of the DID Resolution process -- a DID Method can choose to give you back something totally different.